doc/convolution.md

```diff
@@ -384,13 +384,29 @@ by calling `:ceil()` or `:floor()` methods.
 ### SpatialAveragePooling ###
 
 ```lua
-module = nn.SpatialAveragePooling(kW, kH [, dW, dH])
+module = nn.SpatialAveragePooling(kW, kH [, dW, dH, padW, padH])
 ```
 
 Applies 2D average-pooling operation in `kWxkH` regions by step size
 `dWxdH` steps. The number of output features is equal to the number of
 input planes.
 
+If the input image is a 3D tensor `nInputPlane x height x width`, the output
+image size will be `nOutputPlane x oheight x owidth` where
+
+```lua
+owidth  = op((width  + 2*padW - kW) / dW + 1)
+oheight = op((height + 2*padH - kH) / dH + 1)
+```
+
+`op` is a rounding operator. By default, it is `floor`. It can be changed
+by calling `:ceil()` or `:floor()` methods.
+
+By default, the output of each pooling region is divided by the fixed value
+`kW*kH`. If using padding or `ceil` mode, the number of elements in a region
+can be smaller than `kW*kW`. To switch between different division factors,
```

Comment:
Thanks for pointing out this typo. I'm fixing it.
Also, I should probably slightly reformulate this part of the documentation as well, because now I don't divide by the fixed value `kW*kH` all the time. Indeed, for some degenerate cases (where the end position of the pooling region is outside the padded image), I only count the number of elements inside the padded image, which can be smaller than `kW*kH`